Libraries/Text/TextInput/Singleline/RCTUITextField.m

```diff
@@ -107,6 +109,29 @@ - (CGRect)caretRectForPosition:(UITextPosition *)position
   return [super caretRectForPosition:position];
 }
 
+#pragma mark - Fix for CJK Languages
+// Search for `GH-19339` to see all code affected by this fix.
+
+- (void)setAttributedText:(NSAttributedString *)attributedText {
+  [_attributesHolder setAttributedString:attributedText];
+  [super setAttributedText:attributedText];
+}
+
+- (void)copyAttributesFrom:(NSAttributedString *)source {
+  [_attributesHolder setAttributedString:super.attributedText];
+}
+
+- (NSAttributedString *)attributedText {
```

Comment:
@shergin it is easier to me to comment here, to show why I did it this way.
`UITextField` is a very basic UI element. It doesn't expose any API to just change the string attributes without re-setting the attributed string completely (and breaking CJK input). 
Luckily, it relies on it's own public (and overridable) property `attributedText` when rendering.

What I'm doing here is hijacking this property and replacing its value with the value I store locally, in NSMutableAttributedString, which I can replace without noticeable side-effects. So this override has to be here, if we want to continue to use UITextField. It's a hack and it can potentially break with new iOS releases, etc.

To be able to consolidate logic, we can just use UITextView everywhere, it provides all the necessary functionality as public APIs. What is the history between having 2 different implementations for single and multiline text views?

Reply:
I mean why not consolidate all attribute specific-logic in this class making `attributedText` and `setAttributedText` behave as we want?

We have to have two different underlying components because they simply behave differently. There is no way to configure UITextView behave exactly as UITextField.

Reply:
I mean we have an original problem: When we call `setAttributedText`, market fragment gets reset. So, let's add all checks that prevents that into both  `RCTUITextField` and `RCTUITextView` (iirc, now we have only one of them).

Reply:
ah, okay, I mistread your initial comment.
So you mean, just duplicate the logic and encapsulate it inside `RCTUITextField` and `RCTUITextView` classes separately, so we can call `setAttributedString` safely on both `RCTUITextField` and `RCTUITextView`?

Reply:
should be fixed now

Reply:
(Let's continue discussion here.)
Now we are abstracted from anything else, so we can work on it. 😄 

I think, it's great. ❤️ 

Nits:
* Instead of referring issues and commenting every single line, we have to leave one comment that describe a problem and (most importantly) the change in the behavior of the methods. That's more sustainable and easy to understand for people from the future which don't have a full context.
* Codestyle. (New line before `{` and spaces.)

Btw, I don't think it's only fix for CJK problem. It's more wide problem which happens when we are replacing some attributed string with completely new one without paying attention to what attributes were added to the string by active text input. So, if we can preserve those attributes, I bet full replacing will work with CJK. (The current solution is good enough though.)

Reply:
pushed the fix to address this feedback

Reply:
@shergin is there anything else I need to fix in this PR?